Lib/typing.py

```diff
@@ -2457,6 +2457,19 @@ def __init__(self, name, tp):
         if def_mod != 'typing':
             self.__module__ = def_mod
 
+    def __mro_entries__(cls, bases):
+        new_type_name = cls.__name__
```

Comment:
I find this variable name really confusing. You've called it `new_type_name`, but it's actually the name of the _old_ class -- the class that's being subclassed.

Reply:
I thought new_type_name was alright cause its the name of the `NewType` but I'm open to suggestions

Reply:
Yeah, I get where you got the variable name from! But I kept on tripping up on it when I was playing around with the patch. Why not use `superclass_name`/`subclass_name`? :)